modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -0,0 +1,636 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVRhieChowInterpolator.h"
+#include "INSFVAttributes.h"
+#include "GatherRCDataElementThread.h"
+#include "GatherRCDataFaceThread.h"
+#include "SubProblem.h"
+#include "MooseMesh.h"
+#include "SystemBase.h"
+#include "NS.h"
+#include "Reconstructions.h"
+#include "Assembly.h"
+#include "INSFVVelocityVariable.h"
+#include "INSFVPressureVariable.h"
+#include "PiecewiseByBlockLambdaFunctor.h"
+
+#include "libmesh/mesh_base.h"
+#include "libmesh/elem_range.h"
+#include "libmesh/parallel_algebra.h"
+#include "metaphysicl/dualsemidynamicsparsenumberarray.h"
+#include "metaphysicl/parallel_dualnumber.h"
+#include "metaphysicl/parallel_dynamic_std_array_wrapper.h"
+#include "metaphysicl/parallel_semidynamicsparsenumberarray.h"
+#include "timpi/parallel_sync.h"
+
+using namespace libMesh;
+
+registerMooseObject("NavierStokesApp", INSFVRhieChowInterpolator);
+
+InputParameters
+INSFVRhieChowInterpolator::validParams()
+{
+  auto params = GeneralUserObject::validParams();
+  params += TaggingInterface::validParams();
+  params += BlockRestrictable::validParams();
+  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
+  exec_enum.addAvailableFlags(EXEC_PRE_KERNELS);
+  exec_enum = {EXEC_PRE_KERNELS};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+  params.addRequiredParam<VariableName>(NS::pressure, "The pressure variable.");
+  params.addRequiredParam<VariableName>("u", "The x-component of velocity");
+  params.addParam<VariableName>("v", "The y-component of velocity");
+  params.addParam<VariableName>("w", "The z-component of velocity");
+  params.addParam<bool>("standard_body_forces", false, "Whether to just apply normal body forces");
+  params.addClassDescription("Performs interpolations and reconstructions of body forces and "
+                             "computes the Rhie-Chow face velocities.");
+  return params;
```

Comment:
This object is actually necessary for simulations that just do average velocity interpolation. Maybe I should rename the object to `INSFVInterpolator`? However, I do think that we should probably be removing the ability to perform non-RC-based simulations (for incompressible/weakly-compressible) in the not-too-far future since we've ironed out the kinks in RC and average does nothing but make the simulation more unstable/checkerboardy.

Reply:
I'm down for that.